### PR TITLE
Allow pytorch batching using jax2torch

### PR DESCRIPTION
**Pull request by Giulero**
This PR exploits the package [jax2torch](https://github.com/lucidrains/jax2torch), based on the [gist](https://gist.github.com/mattjj/e8b51074fed081d765d2f3ff90edf0e9) which explains how to convert `jax` functions in `pytorch` ones, preserving also the gradients. 

This should allow to perform batch computations, e.g.
```python
H_b_batch = torch.tile(torch.tensor(H_b), (n_samples, 1, 1)).requires_grad_()
joints_val_batch = torch.tile(torch.tensor(joints_val), (n_samples, 1)).requires_grad_()

mass_matrix = kindyn.mass_matrix(H_b_batch, joints_val_batch)
# The mass matrix dimension should have dimension (n_samples, n_dofs + 6, n_dofs + 6)

# and gradient
mass_matrix.sum().backward()
```

I put this interface in `adam.pytorch` as `KinDynComputionBatch`, even if it using `jax` under the hood.

```python
from adam.pytorch import KinDynComputationsBatch
# and it is instantiated in the vary same way of the other interfaces
kindyn = KinDynComputationsBatch(model_path, joints_name_list)
```

@traversaro already opened https://github.com/conda-forge/staged-recipes/pull/26780#issuecomment-2194579078, so everything will be soon conda ready.

### CODE REVIEW

**Comment by traversaro** [comment]
> @traversaro already opened [conda-forge/staged-recipes#26780 (comment)](https://github.com/conda-forge/staged-recipes/pull/26780#issuecomment-2194579078), so everything will be soon conda ready.

Merged: https://github.com/conda-forge/jax2torch-feedstock .

**Comment by Giulero** [comment]
All tests are passing now!

**Review by traversaro — APPROVED** [review]

**Comment by Giulero** [comment]
Cc @evelyd @Zweisteine96
